main.tf

```diff
@@ -919,6 +919,161 @@ module "aws_load_balancer_controller" {
   tags = var.tags
 }
 
+################################################################################
+# Cluster Autoscaler
+################################################################################
+
+locals {
+  cluster_autoscaler_service_account = try(var.cluster_autoscaler.service_account_name, "external-secrets-sa")
+
+  # Lookup map to pull latest cluster-autoscaler patch version given the cluster version
+  cluster_autoscaler_image_tag = {
```

Comment:
Curious how we get this list and how its maintained?

Reply:
this line does the magic, and its similar to what we have today but now we have a list that provides a better "default" https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/96d7bbdb022dbaa3006ba6325f2e5b4a3a27a365/main.tf#L1034

so it will try to look up a version in that locals list which is just a static list based on the image versions in the CA project 
<img width="465" alt="image" src="https://user-images.githubusercontent.com/10913471/230513159-643a74c0-eabb-49d9-bcdb-5bd3e6398315.png">

the `try(local.cluster_autoscaler_image_tag[var.cluster_version], "v${var.cluster_version}.0")` is trying to get a value from that local map, and if it doesn't fall back to a safe/sane default which is just the release version of CA for a given kubernetes version ( given CA's release version cadence due to the bundled scheduler)